pkg/config/template.go

```diff
@@ -0,0 +1,31 @@
+package config
+
+import (
+	"bytes"
+	"fmt"
+	"text/template"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+func TemplateLabels(labels map[string]string, target *unstructured.Unstructured) (map[string]string, error) {
+	renderedLabels := make(map[string]string, len(labels))
+
+	for key, val := range labels {
+		tmpl, err := template.New("test").Option("missingkey=error").Parse(val)
```

Comment:
I guess we dont need to create a template parser for each key? Can we extract `template.New("test").Option("missingkey=error")` outside of the loop and give it better name than "test"? :)

Reply:
minor optimization, but yeah.. addressed, rebased and updated